### PR TITLE
fixing the wrong result after clicking equal button

### PR DESCRIPTION
**Pull request by yaaHosein**
let us work together !!

### CODE REVIEW

File: CalcualtorLogic.js [hunk 1]
```diff
@@ -171,14 +171,15 @@ class Calculator {
...
-        if (this.lastValue && this.currentResult !== null) {
+        if (this.lastValue && this.currentResult !== null && this.operator !== null) {
             let firstOperand = parseFloat(this.lastValue);
             let secondOperand = parseFloat(this.currentResult);
             let _result;
             if (this.operator === "+") _result = this.calculatorLogic.add(firstOperand, secondOperand);
             if (this.operator === "-") _result = this.calculatorLogic.subtract(firstOperand, secondOperand);
             if (this.operator === "*") _result = this.calculatorLogic.multiply(firstOperand, secondOperand);
             if (this.operator === "/") _result = this.calculatorLogic.divide(firstOperand, secondOperand);
+            if (this.lastValue !== null) this.lastValue === null;
```
Comment (SMH110) [inline CalcualtorLogic.js]: There's no reason for adding this if statement 
also, using `===` is wrong here

` this.lastValue = null ` is enough

File: CalcualtorLogic.js [hunk 2]
```diff
@@ -171,14 +171,15 @@ class Calculator {
         this.updateResultElement();
     }
     evaluate() {
-        if (this.lastValue && this.currentResult !== null) {
+        if (this.lastValue && this.currentResult !== null && this.operator !== null) {
```
Comment (SMH110) [inline CalcualtorLogic.js:174]: You're checking if `operator` is available (not null)
but you're not resetting the operator flag to null 
I would add 
`this.opeator = null `

Comment (yaaHosein) [reply CalcualtorLogic.js:174]: I do not know how the equality changed to ===

I set it to = 
And checked the calculator 
When it is ===
The problem is same problem

Comment (SMH110) [reply CalcualtorLogic.js:174]: is this comment solved ?

**Comment by yaaHosein** [comment]
Yes it is 
If you remember I sent you earlier what I have added regarding to this bug 
But not on this branch 
I think it was 
= null not === null 

Can I close and merge and pull after being on the master b
Thanks

File: CalcualtorLogic.js (hunk 2 above)
Comment (yaaHosein) [reply CalcualtorLogic.js:174]: Yes it is 
So can I merge and close and checkout to master and pull

Thanks

**Comment by yaaHosein** [comment]
I did not close it intentionally

**Comment by SMH110** [comment]
this is not done yet

I said to you that you have check on the operator also
` if (this.lastValue && this.currentResult !== null && this.operator !== null) {` 

but you're not resetting the operator  which make your check almost useless 

you need to rest the operator flag also to null

**Comment by yaaHosein** [comment]
I have pushed the new updates to the same branch
I think it's done
sorry I was thinking you were asking abt     this.lastValue = null;

thanks

**Pull request closed without merging**